packages/generator/src/sdk-metadata/generation-and-usage.ts

```diff
@@ -76,3 +105,106 @@ export function getODataLinks(): Links {
     'OData'
   );
 }
+
+function getODataEntity(
+  serviceName: string,
+  vdmEntities: VdmEntity[]
+): VdmEntity {
+  let closestEntity: VdmEntity | undefined;
+  let minDistance = distanceThreshold;
+
+  // remove special char from service name
+  const serviceNameFormatted = getWordWithoutSpecialChars(serviceName);
+  vdmEntities.forEach(entity => {
+    const distance = getLevensteinDistance(
+      serviceNameFormatted,
+      getWordWithoutSpecialChars(entity.className)
+    );
+    if (distance < minDistance) {
+      minDistance = distance;
+      closestEntity = entity;
+    }
+  });
+
+  if (closestEntity) {
+    return closestEntity!;
+  }
+  // If no closest entity found, return the entity with shortest name
+  return vdmEntities.sort((a, b) =>
+    a.className.length < b.className.length ? -1 : 1
+  )[0];
+}
+
+function getActionFunctionImport(
+  serviceName: string,
+  actionFunctionImports: VdmFunctionImport[] | VdmActionImport[]
+): VdmFunctionImport | VdmActionImport {
+  if (actionFunctionImports.length === 1) {
+    return actionFunctionImports[0];
+  }
+
+  return (
+    getLevensteinClosestFunction(serviceName, actionFunctionImports) ||
+    getFunctionWithoutParameters(actionFunctionImports) ||
+    getFunctionWithMinParameters(actionFunctionImports)
+  );
+}
+
+// Gets function having minimum Levenstein distance with the service name. Returns undefined if distance is greater than threshhold.
+function getLevensteinClosestFunction(
+  serviceName: string,
+  actionFunctionImports: VdmFunctionImport[] | VdmActionImport[]
+): VdmFunctionImport | VdmActionImport | undefined {
+  let closestFunc: VdmFunctionImport | VdmActionImport | undefined;
+  let minDistance = distanceThreshold;
+  actionFunctionImports.forEach(func => {
+    const distance = getLevensteinDistance(serviceName, func.name);
+    if (distance < minDistance) {
+      minDistance = distance;
+      closestFunc = func;
+    }
+  });
+
+  return closestFunc;
+}
+
+function getFunctionWithoutParameters(
+  actionFunctionImports: VdmFunctionImport[] | VdmActionImport[]
+): VdmFunctionImport | VdmActionImport | undefined {
+  const functionsWithoutParams = actionFunctionImports.filter(
+    func => func.parameters?.length === 0
+  );
+  if (functionsWithoutParams) {
+    return functionsWithoutParams[0];
+  }
+}
+
+// Sorts and gets a function import having minimum input parameters.
+function getFunctionWithMinParameters(
+  actionFunctionImports: VdmFunctionImport[] | VdmActionImport[]
+): VdmFunctionImport | VdmActionImport {
+  const getFunctions = actionFunctionImports.filter(
```

Comment:
The method takes minimal parameter but onlu on the getFunctions. Either reflect this in the name or remove it if not needed.